packages/amplify_datastore_plugin_interface/lib/src/types/models/model.dart

```diff
@@ -17,18 +17,14 @@ import 'model_schema.dart';
 import 'model_schema_definition.dart';
 
 abstract class Model {
-  final String id;
-
   ModelType getInstanceType();
 
-  const Model({required this.id});
-
-  String getId() {
-    return id;
-  }
+  String getId();

```

Comment:
Should this throw an UnimplementedError if it no longer has a default implementation?

Reply:
Not a major concern as all models will be generated by codegen team which is guaranteed to have that implementation.

Reply:
Yeah, I consider this to be more of a nit/non-blocking comment. It is hard to imagine a scenario where this would actually be an issue, but if there were some bug in the code generation failing fast by throwing an error would be preferred to returning null.